apepay/__init__.py

```diff
@@ -74,11 +82,67 @@ def validators(self) -> List[Validator]:
                         )
                     )
                 )
-            except ContractLogicError:
+
+            except (ContractLogicError, DecodingError):
+                # NOTE: Vyper returns no data if not a valid index
                 break
 
         return validators
 
+    def _convert_to_address(self, item: _ValidatorItem) -> str:
+        if isinstance(item, Validator):
+            return item.contract.address
+        elif isinstance(item, ContractInstance):
+            return item.address
+        else:
+            return item
+
+    def set_validators(
+        self,
+        validators: List[_ValidatorItem],
+        **txn_kwargs,
+    ) -> ReceiptAPI:
+        if len(validators) > 20:
```

Comment:
Shouldn't this be `>=` 20?

The `def validators()` property implies that you can have up to 20 validators.
this check only ensures it is not > 20 at the start
Thus it will attempt to set the 21st but that one won't show up in the `validators()` list

Reply:
I resolved this, let me know I for some reason am fudging up the understanding tho